packages/myst-cli/src/process/notebook.ts

```diff
@@ -70,13 +74,27 @@ export async function processNotebook(
   file: string,
   content: string,
 ): Promise<GenericParent> {
+  const { mdast } = await processNotebookFull(session, file, content);
+  return mdast;
+}
+
+export async function processNotebookFull(
+  session: ISession,
+  file: string,
+  content: string,
+): Promise<{ mdast: GenericParent; frontmatter: PageFrontmatter }> {
   const { log } = session;
   const { metadata, cells } = JSON.parse(content) as INotebookContent;
   // notebook will be empty, use generateNotebookChildren, generateNotebookOrder here if we want to populate those
 
   const language = metadata?.kernelspec?.language ?? 'python';
   log.debug(`Processing Notebook: "${file}"`);
 
+  // Load frontmatter from notebook metadata
+  const vfile = new VFile();
```

Comment:
Here we are going to loose all error messages as this isn't referenced anywhere. Is that intended? If not, we should pass the vfile in?

Reply:
Hmm, I was thinking that the notebook metadata isn't strictly frontmatter, so it might not always validate. But, if you think that's a strong assumption, then yes we should. Defer to your judgement.

Reply:
There's no test scaffolding right now, does @fwkoch want to weigh in?

Reply:
Regarding tests: for the complicated parts with no test scaffolding, I've been making end-to-end tests in `packages/mystmd/tests` - make a mini project that includes your feature, define the command to run, and expect some output. Seems like that could be possible here?

Regarding the extra keys and extra errors: you could run
```typescript
const filteredMetadata = filterKeys(metadata ?? {}, PAGE_FRONTMATTER_KEYS)
```
before running
```typescript
validatePageFrontmatter(filteredMetadata, ...)
```
then we would get real errors but none of the extra key warnings.

Reply:
@fwkoch I added this and extended it to accept `FRONTMATTER_ALIASES` which are aliases of `PAGE_FRONTMATTER_KEYS`. Does that seem reasonable?

Reply:
Oh, I see you did exactly this in a later commit😆